### ISSUE TITLE
Using incomplete cache key results in duplicate data when querying multiple basecurrencies.

### ISSUE THREAD

**Issue by dan-da** [dan-da/bitsquare_market#8]
I found this problem in the course of investigating:
https://github.com/bisq-network/numbers/issues/1#issuecomment-343261480

A memory cache is used to store results temporarily during a request.  The cache key was not including information about the basecurrency, thus the first result stored for a given key would be used even for other base currencies.

wrong code:
```
        static $results = null;
        $val = @$results[$key];
        if( $val ) {
            return $val;
        }
```

fixed code:
```
        static $results = null;
        $fullkey = $file . $key;
        $val = @$results[$fullkey];
        if( $val ) {
            return $val;
        }
```

**Comment by dan-da** [comment]
fixed in fd377e3211da770e1ac915c5382342f81e680614

**Comment by dan-da** [comment]
related commits:  8dded19a7d290aba2ba94dc101ed358d8631f19c and 64a002e8552cca1828ca60e101e1ea0b19c6830a
